cmd/util/suit.go

```diff
@@ -286,3 +293,35 @@ func ParseNemesisGenerators(names string) (nemesisGens []core.NemesisGenerator)
 	}
 	return
 }
+
+// hasIOChaos checks if there is any io chaos nemesis
+func hasIOChaos(ngs []core.NemesisGenerator) string {
+	for _, v := range ngs {
+		if _, ok := v.(nemesis.IOChaosGenerator); ok {
+			switch v.Name() {
+			case "delay_tikv", "errno_tikv", "mixed_tikv":
+				return "tikv"
+			case "delay_pd", "errno_pd", "mixed_pd":
+				return "pd"
+			}
+		}
+	}
+	return ""
+}
+
+// TODO(yeya24): support other cluster types
+func (suit *Suit) updateClusterDef(ioChaosType string) {
+	switch s := suit.ClusterDefs.(type) {
+	case *tidb.TiDBClusterRecommendation:
+		if ioChaosType == "tikv" {
+			s.TidbCluster.Spec.TiKV.Annotations = map[string]string{
+				"admission-webhook.pingcap.com/request": "chaosfs-tikv",
+			}
+		} else if ioChaosType == "pd" {
+			s.TidbCluster.Spec.PD.Annotations = map[string]string{
+				"admission-webhook.pingcap.com/request": "chaosfs-pd",
+			}
+		}
+		suit.ClusterDefs = s
```

Comment:
This assignment is redundant?

Reply:
Thanks. You are right. I have updated it.